bootstrap/internal/controller/agent_controller.go

```diff
@@ -82,11 +82,37 @@ func (r *AgentReconciler) setAgentFields(ctx context.Context, agent *aiv1beta1.A
 	agent.Spec.Role = role
 	agent.Spec.IgnitionConfigOverrides = ignitionConfigOverrides
 
-	// TODO: make sure we won't approve if an Agent with the same infraenv is already around
-	agent.Spec.Approved = true
+	approvable, err := r.canApproveAgent(ctx, agent)
+	if err != nil {
+		return err
+	}
+
+	agent.Spec.Approved = approvable
 	return r.Client.Update(ctx, agent)
 }
 
+func (r *AgentReconciler) canApproveAgent(ctx context.Context, agent *aiv1beta1.Agent) (bool, error) {
+	agentList := &aiv1beta1.AgentList{}
+	if err := r.Client.List(ctx, agentList, client.InNamespace(agent.Namespace)); err != nil {
```

Comment:
Can you use the infraenv of the existing agent in the list params here to simplify this?